src/Storages/ObjectStorage/Azure/Configuration.h

```diff
@@ -58,8 +58,8 @@ class StorageAzureConfiguration : public StorageObjectStorage::Configuration
     Path getPath() const override { return blob_path; }
     void setPath(const Path & path) override { blob_path = path; }
 
-    const Paths & getPaths() const override { return blobs_paths; }
-    void setPaths(const Paths & paths) override { blobs_paths = paths; }
+    const DataFileInfos & getPaths() const override { return blobs_paths; }
```

Comment:
I don't really like this change tbh. `DataFileInfos` contains some specific information that is used only in Iceberg engine and not really related to usual Object storages/configuration. I would like to keep logic of Iceberg as isolated as possible. (Ideally, it would be good to have separate `StorageIceberg` with `StorageObjectStorage` as a base class so we can implement all custom logic for Iceberg in `StorageIceberg` using virtual methods. But maybe for now it's too much and we can do it later.

Right now as I understand we need to support files with different schemas and make schemas transformation for each file after reading. Maybe, instead of having schema information stored together with paths, we can store a separate map `std::unordered_map<String, FileSchemaChangesInfo> files_schema_changes_info` in `IcebergMetadata` next to `Strings data_files` and fill it only for files with changes in schema. And we can propogate this `files_schema_changes_info` together with `read_from_format_info` to `StorageObjectStorageSource` and use it there. We already prepare `read_from_format_info` separately in `IStorageDataLake::prepareReadingFromFormat` and can make similar virtual method `IStorageDataLake::getFilesSchemaChangesInfo`. And it will isolate these changes a lot and we won't need to change code in configurations and minimize changes in `StorageObjectStorage`.

WDYT?

Reply:
Yes, I think this approach is better, I will change the logic

Reply:
We created a big refactoring which on contrary made StorageObjectStorage closer to Data Lakes. Though I think that your comment is still relevant. I do not think that propagating files_schema_changes_info to StorageObject Storage is a good idea, because only one entity should be aware of the changed schema. But now we have super configuration class, I will use it for getting all the changed info about files